.gitlab-ci-full-integration.yml

```diff
@@ -0,0 +1,48 @@
+test:integration:
+  rules:
+    - if: $RUN_TESTS_FULL_INTEGRATION == "true"
+  image: docker:${DOCKER_VERSION}-dind
+  stage: test
+  tags:
+    - mender-qa-worker-integration-tests
+  timeout: 24h
+  variables:
+    PYTEST_XDIST_AUTO_NUM_WORKERS: "4"
+  before_script:
+    - |
+      docker login registry.mender.io \
+      --username $REGISTRY_MENDER_IO_USERNAME \
+      --password $REGISTRY_MENDER_IO_PASSWORD
+    - |
+      docker login docker.io \
+      --username $DOCKER_HUB_USERNAME \
+      --password $DOCKER_HUB_PASSWORD
```

Comment:
This won't work on on regular PRs (your branch is protected). How are we planning to solve it?

Edit: I see below that we plan to run these tests only on protected branches - that won't work when we develop new tests or debug existing ones.